### PR TITLE
Improvement to project files using Directory.Build.props

### PR DESCRIPTION
**Pull request by huoyaoyuan**
*I'm working on my own project these days, and found some noticeable things.*

The special name `Directory.Build.props` will be automatically imported by latest MSBuild.
Common property file should use relative path from itself.
The service guid is for unit test, so it's sure to keep it only in test project. (If ci is happy for this)

###### Justification:
Should latest language version be introduced? C# releases minor versions rapidly now, and third-party tools may not understand it, especially for non-Visual Studio users.
But C# 7.x has improved some performance (span and readonly struct), making minor version update worth while.

Roslyn team is interested in enforcing code style at build time. Should I introduce rules in editor config now?

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <!-- Contains required properties for osu!framework projects. -->
 <Project>
   <PropertyGroup Label="C#">
-    <LangVersion>7</LangVersion>
+    <LangVersion>7.3</LangVersion>
```
Comment (smoogipoo) [inline Directory.Build.props]: We are not ready to jump into 7.3 just yet. Not supported on the latest mono preview even.

**Comment by smoogipoo** [comment]
I don't know if we'll be using roslyn analysers going forward, they need more discussion. 👎  for now.

File: Directory.Build.props (hunk 1 above)
Comment (huoyaoyuan) [reply Directory.Build.props]: JetBrains supports 7.2 about a month ago, but Mono support for 7.2 is still in preview. So revert language version to 7.0, and keep other changes?

Comment (smoogipoo) [reply Directory.Build.props]: Revert to 7.0 for now I'd say.

**Comment by smoogipoo** [comment]
Cool stuff

**Pull request merged**